gpytorch/mlls/gamma_robust_variational_elbo.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import math
+import torch
+from ._approximate_mll import _ApproximateMarginalLogLikelihood
+from ..likelihoods import _GaussianLikelihoodBase
+
+
+class GammaRobustVariationalELBO(_ApproximateMarginalLogLikelihood):
+    r"""
+    An alternative to the variational evidence lower bound (ELBO), proposed by `Knoblauch, 2019`_.
+    It is derived by replacing the log-likelihood term in the ELBO with a `\gamma` divergence:
+
+    .. math::
+
+       \begin{align*}
+          \mathcal{L}_{\gamma} &=
+          \sum_{i=1}^N \mathbb{E}_{q( \mathbf u)} \left[
+            -\frac{\gamma}{1 - \gamma}
+            \frac{
+                p( y_i \! \mid \! \mathbf u)^{\gamma - 1}
+            }{
+                \int p(\mathbf y \mid \mathbf u) \: d \mathbf y
+            }
+          \right] - \beta \: \text{KL} \left[ q( \mathbf u) \Vert p( \mathbf u) \right]
+       \end{align*}
+
+    where :math:`N` is the amount of data, :math:`\gamma` is a hyperparameter,
+    :math:`q(\mathbf u)` is the variational distribution for
+    the inducing function values, and `p(\mathbf u)` is the prior distribution for the inducing function
+    values.
+
+    :math:`\beta` is a scaling constant for the KL divergence.
+
+    Args:
+        :attr:`likelihood` (:obj:`gpytorch.likelihoods.Likelihood`):
+            The likelihood for the model
+        :attr:`model` (:obj:`gpytorch.models.ApproximateGP`):
+            The approximate GP model
+        :attr:`num_data` (int):
+            The total number of training data points (necessary for SGD)
+        :attr:`beta` (float - default 1.):
+            A multiplicative factor for the KL divergence term.
+        :attr:`gamma` (float - default 1.):
+            Controls the :math:`\gamma` term in the robust ELBO.
+        :attr:`combine_terms` (bool):
+            Whether or not to sum the expected NLL with the KL terms (default True)
+
+    Example:
+        >>> # model is a gpytorch.models.VariationalGP
+        >>> # likelihood is a gpytorch.likelihoods.Likelihood
+        >>> mll = gpytorch.mlls.VariationalELBO(likelihood, model, num_data=100, beta=0.5)
+        >>>
+        >>> output = model(train_x)
+        >>> loss = -mll(output, train_y)
+        >>> loss.backward()
+
+    .. _Knoblauch, 2019:
+        https://arxiv.org/pdf/1904.02303.pdf
+    .. _Knoblauch, Jewson, Damoulas 2019:
+        https://arxiv.org/pdf/1904.02063.pdf
+    """
+    def __init__(self, likelihood, model, gamma=1.0, *args, **kwargs):
```

Comment:
Make sure that we're consistent with what value of gamma we're using (either `gamma - 1` or `gamma`). Typical value should be 1.03